server/activate_hooks.go

```diff
@@ -3,41 +3,59 @@ package main
 import (
 	"fmt"
 
+	"github.com/blang/semver"
 	"github.com/mattermost/mattermost-server/model"
 )
 
+const minimumServerVersion = "5.4.0"
+
 // OnActivate is invoked when the plugin is activated.
 //
 // This demo implementation logs a message to the demo channel whenever the plugin is activated.
 func (p *Plugin) OnActivate() error {
-	teams, err := p.API.GetTeams()
+	v, err := semver.Parse(p.API.GetServerVersion())
 	if err != nil {
 		p.API.LogError(
-			"failed to query teams OnActivate",
+			"failed to parse server version",
 			"error", err.Error(),
 		)
+		return err
+	}
+	r := semver.MustParseRange(">=" + minimumServerVersion)
+	if !r(v) {
+		return fmt.Errorf("current Mattermost version is to low. Please update your Mattermost Server to a least v%s.", minimumServerVersion)
+	}
+
+	teams, appErr := p.API.GetTeams()
+	if appErr != nil {
+		p.API.LogError(
+			"failed to query teams OnActivate",
+			"error", appErr.Error(),
+		)
+		return appErr
 	}
 
 	for _, team := range teams {
-		if _, err := p.API.CreatePost(&model.Post{
+		_, appErr := p.API.CreatePost(&model.Post{
```

Comment:
@hanzei, is this the idiomatic error handling changes to which you referred? The code as was written was already idiomatic. The changes we've been making in the main codebase and elsewhere are to fix blocks like this:
```go
if x, err := DoSomething(); err != nil {
    // error handling
    return err
} else {
    // do something with x

    if y, err := DoSomethingElse(); err != nil {
        // ...
    }
}
```

to avoid the nesting and return early:
```go
x, err := DoSomething(); 
if err != nil {
    // error handling
    return err
}

// do something with x

y, err := DoSomethingElse(); 
if err != nil {
        // ...
}
```

If we're not using the return value of the function outside the error handling block, it's fine to declare them inline in the conditional and handle immediately:
```go
if err := DoSomethingLast(); err != nil {
    // error handling
    return err
}
```

Reply:
Thanks for the clarification! I reverted my changes.